### PR TITLE
[Custom] Add an example scaffolding

### PR DESCRIPTION
**Pull request by zhoonit**
- ~**#745** [AppContext] Add registerer,invoke factory methods~
- ~**#746** [AppContext] Register Default ops at the begining~
- ~**#753** [AppContext] Fix key is case sensitive~
- ~**#753** [Layer] Add built-in ops to the context~
- [Custom] Add an example scaffolding

### CODE REVIEW

**Review by jijoongmoon — APPROVED** [review]
LGTM

File: Applications/Custom/LayerClient/jni/Android.mk [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ifndef ANDROID_NDK
+$(error ANDROID_NDK is not defined!)
+endif
+
+ifndef NNTRAINER_ROOT
+NNTRAINER_ROOT := $(LOCAL_PATH)/../../../..
+endif
+
+NNTRAINER_INCLUDES := $(NNTRAINER_ROOT)/nntrainer \
+	$(NNTRAINER_ROOT)/nntrainer/dataset \
```
Comment (kparichay) [inline Applications/Custom/LayerClient/jni/Android.mk:15]: is this intended? client wont have access to these headers

File: Applications/Custom/LayerClient/jni/main.cpp [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  }
+
+  return ML_ERROR_NONE;
+}
+
+int main(int argc, char *argv[]) {
+  /**< add argc */
+  auto model = ml::train::createModel(ml::train::ModelType::NEURAL_NET);
+
+  model->loadFromConfig("..model.ini");
```
Comment (kparichay) [inline Applications/Custom/LayerClient/jni/main.cpp:62]: `../model.ini`?

File: Applications/Custom/LayerClient/jni/pow.h [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+                   const std::string &value = "") {
+    /**< NOT USED */
+  }
+
+  /**
+   * @brief check if hyperparameter is valid
+   *
+   * @return int ML_ERROR_NONE if successful
+   */
+  int checkValidataion() { return ML_ERROR_NONE; }
```
Comment (kparichay) [inline Applications/Custom/LayerClient/jni/pow.h:64]: `checkValidation`

File: Applications/Custom/LayerClient/jni/Android.mk (hunk 1 above)
Comment (kparichay) [reply Applications/Custom/LayerClient/jni/Android.mk:15]: If for now, not exposed to API, then this is fine.

Comment (zhoonit) [reply Applications/Custom/LayerClient/jni/Android.mk:15]: Yes, I'll add todo here as well (in 779)

File: Applications/Custom/LayerClient/jni/pow.h (hunk 3 above)
Comment (zhoonit) [reply Applications/Custom/LayerClient/jni/pow.h:64]: This is fixed by inheriting layer_internal.h and the function is deleted.

**Review by kparichay — APPROVED** [review]
LGTM

**Pull request merged**